network/simple-onchain-discovery/src/lib.rs

```diff
@@ -61,106 +59,63 @@ pub fn gen_simple_discovery_reconfig_subscription(
-    account: &AccountAddress,
-    addr_idx: u32,
-    raw_enc_addr: RawEncNetworkAddress,
-) -> Result<RawNetworkAddress> {
-    let enc_addr = EncNetworkAddress::try_from(&raw_enc_addr)
-        .map_err(anyhow::Error::new)
-        .with_context(|| {
-            format_err!(
-                "error deserializing encrypted network address: {:?}",
-                raw_enc_addr
-            )
-        })?;
-
-    let key_version = enc_addr.key_version();
-    let shared_val_netaddr_key = shared_val_netaddr_key_map
-        .get(&key_version)
-        .ok_or_else(|| format_err!("no shared_val_netaddr_key for version: {}", key_version))?;
-    let raw_addr = enc_addr.decrypt(shared_val_netaddr_key, account, addr_idx)?;
-    Ok(raw_addr)
-}
-
 /// Extracts a set of ConnectivityRequests from a ValidatorSet which are appropriate for a network with type role.
 fn extract_updates(
     role: RoleType,
     shared_val_netaddr_key_map: &HashMap<KeyVersion, Key>,
     node_set: ValidatorSet,
 ) -> Vec<ConnectivityRequest> {
-    // TODO(philiphayes): remove this after removing explicit pubkey field
-    let explicit_pubkeys: HashMap<_, _> = node_set
-        .payload()
-        .iter()
-        .map(|info| {
-            let peer_id = *info.account_address();
-            let pubkey = match role {
-                RoleType::Validator => info.config().validator_network_identity_public_key,
-                RoleType::FullNode => info.config().full_node_network_identity_public_key,
-            };
-            (peer_id, pubkey)
-        })
-        .collect();
-
     // Collect the set of address updates.
     let new_peer_addrs: HashMap<_, _> = node_set
         .into_iter()
         .map(|info| {
             let peer_id = *info.account_address();
             let config = info.into_config();
-            // only one field currently, though this will change soon
-            let addr_idx = 0;
 
-            let raw_addr_res = match role {
+            let addrs: Vec<NetworkAddress> = match role {
                 RoleType::Validator => {
-                    let raw_enc_addr = config.validator_network_address;
-                    decrypt_validator_netaddr(
-                        &shared_val_netaddr_key_map,
+                    // decrypt and deserialize validator network addresses.
+                    // log and skip those that fail for whatever reason.
+                    decrypt_addresses(
                         &peer_id,
-                        addr_idx,
-                        raw_enc_addr,
+                        &shared_val_netaddr_key_map,
+                        &config.validator_network_addresses,
                     )
+                    .filter_map(|maybe_addr| {
+                        maybe_addr
+                            .map_err(|err| {
+                                warn!("Failed to read validator network address: {}", err)
+                            })
+                            .ok()
```

Comment:
Does this map the errors, and then just check if there wasn't an error?  I think you can remove the map_err